src/functional/ffront/ast_passes/rewrite_lineno.py

```diff
@@ -0,0 +1,72 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+import ast
+import copy
+import dataclasses
+
+
+@dataclasses.dataclass(kw_only=True)
+class RewriteLineNumbers(ast.NodeTransformer):
+    """
+    AST pass transforming line numbers by adding a fixed offset.
+
+    This pass is useful to make line numbers consistent with the source file
+    an AST node originated from.
+    """
+
+    starting_line: int
+    inherit_from_parent: bool
+
+    _parent_node: ast.AST
+
+    @classmethod
+    def apply(
+        cls,
+        node: ast.AST,
+        starting_line: int,
+        *,
+        inherit_from_parent: bool = True,
+        inplace: bool = False,
```

Comment:
Just a suggestion, since utilities in `ast` always mutate nodes in place, I don't think the `inplace` option here is needed. In the rare cases where the caller does not want modifications in place, he can call directly to `copy.deepcopy()`.